libkbfs/tlf_journal.go

```diff
@@ -476,25 +517,30 @@ func (j *tlfJournal) doBackgroundWorkLoop(
 				j.log.CDebugf(ctx,
 					"Got pause signal for %s", j.tlfID)
 				bws = TLFJournalBackgroundWorkPaused
+				j.pause(journalPausedFromSignal)
 
 			case <-j.needShutdownCh:
 				j.log.CDebugf(ctx,
 					"Got shutdown signal for %s", j.tlfID)
 				needShutdown = true
 			}
 
-			errCh = nil
 			// Cancel the worker goroutine as we exit this
 			// state.
 			bwCancel()
 			bwCancel = nil
+
+			// Ensure the worker finishes after being canceled, so it
```

Comment:
This comment is a bit confusing -- how can the work pick up any new work once it's canceled?